### PR TITLE
Init

### PR DESCRIPTION
**Pull request by ybs64**

### CODE REVIEW

**Comment by ybs64** [comment]
## Danger has errored
[!] Invalid `Dangerfile` file: undefined local variable or method `app` for #\<Danger::Dangerfile:0x000055839ceec760\>
```
 #  from Dangerfile:3
 #  -------------------------------------------
 #  android_lint.filtering = true
 >  android_lint.report_file = app/build/reports/lint-results.xml
 #  android_lint.lint(inline_mode: true)
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.example.prototype">
+
+    <application
```
Comment (ybs64) [inline app/src/main/AndroidManifest.xml:5]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ybs64/Test/blob/4b6133523aafe61eda4410c4cd3e3e5a6e986d55/app/src/main/AndroidManifest.xml#L5"></span>On SDK version 23 and up, your app data will be automatically backed up and restored on app install. Consider adding the attribute <code>android:fullBackupContent</code> to specify an <code>@xml</code> resource which configures which files to backup. More info: https://developer.android.com/training/backup/autosyncapi.html</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/example/prototype/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        switch(1) {
+            default:
+                break;
+            case 1:
+        }
+
+        AsyncTask async = new AsyncTask() {
```
Comment (ybs64) [inline app/src/main/java/com/example/prototype/MainActivity.java:21]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ybs64/Test/blob/4b6133523aafe61eda4410c4cd3e3e5a6e986d55/app/src/main/java/com/example/prototype/MainActivity.java#L21"></span>This <code>AsyncTask</code> class should be static or leaks might occur (anonymous android.os.AsyncTask)</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/example/prototype/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+public class MainActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        switch(1) {
+            default:
```
Comment (ybs64) [inline app/src/main/java/com/example/prototype/MainActivity.java:16]: CRITICAL Move this default to the end of the switch. [rule](http://192.168.11.19:9000/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)

File: app/src/main/java/com/example/prototype/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+                break;
```
Comment (ybs64) [inline app/src/main/java/com/example/prototype/MainActivity.java:17]: MINOR Make this line start at column 9. [rule](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AIndentationCheck)

File: app/src/main/java/com/example/prototype/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+            case 1:
+        }
+
+        AsyncTask async = new AsyncTask() {
+            @Override
+            protected Object doInBackground(Object[] objects) {
```
Comment (ybs64) [inline app/src/main/java/com/example/prototype/MainActivity.java:23]: MINOR Make this line start at column 11. [rule](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AIndentationCheck)
MINOR Move this left curly brace to the beginning of next line of code. [rule](http://192.168.11.19:9000/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

File: app/src/main/java/com/example/prototype/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package com.example.prototype;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.os.AsyncTask;
+import android.os.Bundle;
+
+public class MainActivity extends AppCompatActivity {
```
Comment (ybs64) [inline app/src/main/java/com/example/prototype/MainActivity.java:8]: MINOR Move this left curly brace to the beginning of next line of code. [rule](http://192.168.11.19:9000/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

File: app/src/main/java/com/example/prototype/MainActivity.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+        switch(1) {
+            default:
+                break;
+            case 1:
+        }
+
+        AsyncTask async = new AsyncTask() {
+            @Override
+            protected Object doInBackground(Object[] objects) {
+                return null;
```
Comment (ybs64) [inline app/src/main/java/com/example/prototype/MainActivity.java:24]: MINOR Make this line start at column 13. [rule](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AIndentationCheck)

File: app/src/main/java/com/example/prototype/MainActivity.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.os.AsyncTask;
+import android.os.Bundle;
+
+public class MainActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```
Comment (ybs64) [inline app/src/main/java/com/example/prototype/MainActivity.java:11]: MINOR Make this line start at column 3. [rule](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AIndentationCheck)
MINOR Move this left curly brace to the beginning of next line of code. [rule](http://192.168.11.19:9000/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

File: app/src/main/java/com/example/prototype/MainActivity.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+        super.onCreate(savedInstanceState);
```
Comment (ybs64) [inline app/src/main/java/com/example/prototype/MainActivity.java:12]: MINOR Make this line start at column 5. [rule](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AIndentationCheck)

File: app/src/main/java/com/example/prototype/MainActivity.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+        setContentView(R.layout.activity_main);
+
+        switch(1) {
```
Comment (ybs64) [inline app/src/main/java/com/example/prototype/MainActivity.java:15]: MINOR Replace this "switch" statement by "if" statements to increase readability. [rule](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS1301)
MINOR Move this left curly brace to the beginning of next line of code. [rule](http://192.168.11.19:9000/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

**Comment by ybs64** [comment]
SonarQube analysis reported 15 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 1 critical
* ![MINOR][MINOR] 13 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [MainActivity.java#](https://github.com/ybs64/Test/blob/4b6133523aafe61eda4410c4cd3e3e5a6e986d55/app/src/main/java/com/example/prototype/MainActivity.java#): Add or update the header of this file. [rule](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS1451)
1. ![MINOR][MINOR] [prototype#](https://github.com/ybs64/Test/blob/4b6133523aafe61eda4410c4cd3e3e5a6e986d55/app/src/main/java/com/example/prototype#): Add a 'package-info.java' file to document the 'prototype' package [rule](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [MainActivity.java#](https://github.com/ybs64/Test/blob/4b6133523aafe61eda4410c4cd3e3e5a6e986d55/app/src/main/java/com/example/prototype/MainActivity.java#): Add a new line at the end of this file. [rule](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS00113)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (ybs64) [inline app/src/main/AndroidManifest.xml:5]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ybs64/Test/blob/68ba67a69872065cf9193a315bbf5835fc6c7f40/app/src/main/AndroidManifest.xml#L5"></span>On SDK version 23 and up, your app data will be automatically backed up and restored on app install. Consider adding the attribute <code>android:fullBackupContent</code> to specify an <code>@xml</code> resource which configures which files to backup. More info: https://developer.android.com/training/backup/autosyncapi.html</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/example/prototype/MainActivity.java (hunk 2 above)
Comment (ybs64) [inline app/src/main/java/com/example/prototype/MainActivity.java:21]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ybs64/Test/blob/68ba67a69872065cf9193a315bbf5835fc6c7f40/app/src/main/java/com/example/prototype/MainActivity.java#L21"></span>This <code>AsyncTask</code> class should be static or leaks might occur (anonymous android.os.AsyncTask)</td>
    </tr>
  </tbody>
</table>

File: app/build.gradle [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    lintOptions {
+        checkAllWarnings true
+    }
+}
+
+dependencies {
+    implementation fileTree(dir: "libs", include: ["*.jar"])
+    implementation 'androidx.appcompat:appcompat:1.1.0'
+    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
+    testImplementation 'junit:junit:4.12'
```
Comment (ybs64) [inline app/build.gradle:33]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ybs64/Test/blob/710e3291836c9205962d379297e5ab48dd6a941e/app/build.gradle#L33"></span>A newer version of junit:junit than 4.12 is available: 4.13</td>
    </tr>
  </tbody>
</table>

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (ybs64) [inline app/src/main/AndroidManifest.xml:5]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ybs64/Test/blob/710e3291836c9205962d379297e5ab48dd6a941e/app/src/main/AndroidManifest.xml#L5"></span>App is not indexable by Google Search; consider adding at least one Activity with an ACTION-VIEW intent filter. See issue explanation for more details.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/example/prototype/MainActivity.java (hunk 8 above)
Comment (ybs64) [inline app/src/main/java/com/example/prototype/MainActivity.java:11]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ybs64/Test/blob/710e3291836c9205962d379297e5ab48dd6a941e/app/src/main/java/com/example/prototype/MainActivity.java#L11"></span>Unknown nullability; explicitly declare as <code>@Nullable</code> or <code>@NonNull</code> to improve Kotlin interoperability; see https://android.github.io/kotlin-guides/interop.html#nullability-annotations</td>
    </tr>
  </tbody>
</table>

**Comment by ybs64** [comment]
SonarQube analysis reported 2 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by ybs64** [comment]
Can one of the admins verify this patch?

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (ybs64) [inline app/src/main/AndroidManifest.xml:5]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ybs64/Test/blob/a6cb2f3c790cb6ded2f8890f2c12ca33be8277f2/app/src/main/AndroidManifest.xml#L5"></span>On SDK version 23 and up, your app data will be automatically backed up and restored on app install. Consider adding the attribute <code>android:fullBackupContent</code> to specify an <code>@xml</code> resource which configures which files to backup. More info: https://developer.android.com/training/backup/autosyncapi.html</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/example/prototype/MainActivity.java (hunk 2 above)
Comment (ybs64) [inline app/src/main/java/com/example/prototype/MainActivity.java:21]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ybs64/Test/blob/a6cb2f3c790cb6ded2f8890f2c12ca33be8277f2/app/src/main/java/com/example/prototype/MainActivity.java#L21"></span>This <code>AsyncTask</code> class should be static or leaks might occur (anonymous android.os.AsyncTask)</td>
    </tr>
  </tbody>
</table>

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (ybs64) [inline app/src/main/AndroidManifest.xml:5]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ybs64/Test/blob/94739c36da2d3ee265b095a2bb9622564a439126/app/src/main/AndroidManifest.xml#L5"></span>On SDK version 23 and up, your app data will be automatically backed up and restored on app install. Consider adding the attribute <code>android:fullBackupContent</code> to specify an <code>@xml</code> resource which configures which files to backup. More info: https://developer.android.com/training/backup/autosyncapi.html</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/example/prototype/MainActivity.java (hunk 2 above)
Comment (ybs64) [inline app/src/main/java/com/example/prototype/MainActivity.java:21]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ybs64/Test/blob/94739c36da2d3ee265b095a2bb9622564a439126/app/src/main/java/com/example/prototype/MainActivity.java#L21"></span>This <code>AsyncTask</code> class should be static or leaks might occur (anonymous android.os.AsyncTask)</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/example/prototype/MainActivity.java (hunk 3 above)
Comment (ybs96) [inline app/src/main/java/com/example/prototype/MainActivity.java:16]: CRITICAL Move this default to the end of the switch. [rule](http://192.168.11.19:9000/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)

File: app/src/main/java/com/example/prototype/MainActivity.java (hunk 10 above)
Comment (ybs96) [inline app/src/main/java/com/example/prototype/MainActivity.java:15]: MINOR Replace this "switch" statement by "if" statements to increase readability. [rule](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS1301)

**Comment by ybs96** [comment]
SonarQube analysis reported 2 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request closed without merging**